### PR TITLE
[B2BP-169] - Terraform SSM AWS

### PR DESCRIPTION
**Pull request by maurodandrea**
Creation of Terraform scripts for the automatic deployment on AWS of SSM Parameter Store necessary for Strapi

#### List of Changes
Creation of Terraform scripts for the automatic deployment on AWS of SSM Parameter Store necessary for Strapi

#### Motivation and Context
As reported in the [RFC "SV-003 - Architettura Strapi on AWS"](https://pagopa.atlassian.net/wiki/spaces/SV/pages/795771078/SV-003+-+Architettura+Strapi+on+AWS), to host the Strapi instance on the AWS Cloud Provider it is necessary to create the relevant resources using Terraform scripts, such as: Strapi Server (Cluster ECS Fargate), Database RDS Aurora PostgreSQL, Bucket S3 for Media Library, Load Balancer, Role IAM and Secrets SSM)

#### How Has This Been Tested?
Running Terraform scripts on locally PC pointing to a test AWS environment

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change not requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .infrastructure/04_ssm.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "aws_ssm_parameter" "admin_jwt_secret" {
+  name  = "/website-strapi/admin_jwt_secret"
+  type  = "SecureString"
+  value = random_password.jwt_secret.result
+}
```
Comment (datalek) [inline .infrastructure/04_ssm.tf:23]:
```suggestion
resource "random_password" "cms_database_password" {
  length           = 16
  special          = true
  override_special = "!#$%&*()-_=+[]{}<>:?"
}

resource "aws_ssm_parameter" "cms_database_password" {
  name  = "/cms/database/password"
  type  = "SecureString"
  value = random_password.cms_database_password.result
}

resource "random_password" "cms_admin_jwt_secret" {
  length           = 16
  special          = true
  override_special = "!#$%&*()-_=+[]{}<>:?"
}

resource "aws_ssm_parameter" "cms_admin_jwt_secret" {
  name  = "/cms/admin_jwt_secret"
  type  = "SecureString"
  value = random_password.cms_admin_jwt_secret.result
}
```

**Review by datalek — CHANGES_REQUESTED** [review]
How does `ssm` handle secrets rotation?

**Comment by maurodandrea** [comment]
> How does `ssm` handle secrets rotation?

SSM Parameter Store doesn't provide automatic rotation services for stored secrets.
https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_parameterstore.html
Automatic rotation exists for secrets (generated and managed) by some AWS services, such as RDS Aurora (not JWT_secret). But the problem will be that the Strapi container will go into error when RDS changes atumatically the password (passed through the environment var) if the relevant service will not take the updated password is not restarted. It is best to do an automatic rotation using lambda which updates the password and restarts the ECS service.

File: .infrastructure/04_ssm.tf
Comment (datalek) [inline .infrastructure/04_ssm.tf:1]: What do you think to rename the file to `04_secrets.tf`?

**Review by datalek — APPROVED** [review]

File: .infrastructure/04_ssm.tf
Comment (maurodandrea) [reply .infrastructure/04_ssm.tf:1]: done

**Review by datalek — APPROVED** [review]

**Pull request merged**